### PR TITLE
More core hpp doc corrections

### PR DESCRIPTION
**Pull request by jtlap**
these hpp files still have documentation lacks or malformations or perhaps are not in the right place

broadcast                      is not in proper format
broadcast_group                is not in proper format
combine                        is not in proper format
convert                        is not in proper format
deinterleave_groups            is not in proper format
deinterleave_groups_shuffle    is not in proper format
first_true                     is not in proper format
gather                         is not in proper format
has_equal_in                   is not in proper format
has_single_bit                 has no defined doxygen group
interleave                     has no defined doxygen group
interleave_shuffle             has no defined doxygen group
iterate_selected               is not in proper format
last_true                      has no defined doxygen group
load                           is not in proper format
lookup                         has no defined doxygen group
read                           is not in proper format
reduce                         is not in proper format
replace                        is not in proper format
rotate                         is not in proper format
safe                           has no defined doxygen group
scan                           is not in proper format
scatter                        is not in proper format
shuffle                        has no defined doxygen group
shuffle_v2                     has no defined doxygen group
slide_left                     has no defined doxygen group
slide_right                    has no defined doxygen group
splat                          is not in proper format
store                          is not in proper format
store_equivalent               is not in proper format
swap_if                        is not in proper format
swap_pairs                     is not in proper format
try_each_group_position        is not in proper format
unalign                        is not in proper format
unsafe                         has no defined doxygen group
write                          is not in proper format

### CODE REVIEW

File: include/eve/module/core/regular/all.hpp [hunk 1]
```diff
@@ -30,8 +30,8 @@ namespace eve
 //!   namespace eve
 //!   {
 //!      // Regular overloads
-//!      constexpr bool all(value auto x)                                              noexcept; // 1
-//!      constexpr bool all(top_bits auto M t)                                         noexcept; // 1
+//!      constexpr auto all(logical_value auto x)                                      noexcept; // 1
```
Comment (jfalcou) [inline include/eve/module/core/regular/all.hpp:33]: No, all do returns a bool

File: include/eve/module/core/regular/any.hpp [hunk 2]
```diff
@@ -30,31 +30,25 @@ namespace eve
 //!   namespace eve
 //!   {
 //!      // Regular overloads
-//!      constexpr bool any(value auto x)                                              noexcept; // 1
-//!      constexpr bool any(top_bits auto M t)                                         noexcept; // 1
+//!      constexpr auto any(logical_value auto x)                                      noexcept; // 1
```
Comment (jfalcou) [inline include/eve/module/core/regular/any.hpp:33]: Same, it does alwasy return bool

File: include/eve/module/core/regular/bit_select.hpp [hunk 3]
```diff
@@ -42,13 +42,14 @@ namespace eve
 //!   @code
 //!   namespace eve
 //!   {
-//!     template < value M, value T, value U > constexpr auto bit_select((Logical<M> m, T x, U y) noexcept;
+//!      // Regular overload
+//!      constexpr auto bit_select((auto Logical_value auto  m, auto value x, auto value y) noexcept;
```
Comment (jfalcou) [inline include/eve/module/core/regular/bit_select.hpp:46]: why Logical_value when you say m is a value later ?

File: include/eve/module/core/regular/count_true.hpp [hunk 4]
```diff
@@ -28,16 +28,27 @@ namespace eve
 //!   @code
 //!   namespace eve
 //!   {
-//!      template< eve::value T >
-//!      T count_true(T x) noexcept;                 //1
-//!   }
-//!      template< eve::top_bits M >
-//!      as_wide_as<unsigned, M> any(M m) noexcept;  //2
+//!      // Regular overloads
+//!      constexpr auto count_true(logical_value auto x)                                      noexcept; // 1
```
Comment (jfalcou) [inline include/eve/module/core/regular/count_true.hpp:32]: Why are we putting auto when , before, we had proper type informations.

File: include/eve/module/core/regular/dot.hpp [hunk 5]
```diff
@@ -42,23 +42,22 @@ namespace eve
...
 //!
-//!     * `x`, `y`  :  [real arguments](@ref eve::value).
+//!     * `x`, `y`  :  [value arguments](@ref eve::value).
+//!     * `c`: [Conditional expression](@ref conditional_expr) masking the operation.
+//!     * `m`: [Logical value](@ref logical) masking the operation.
 //!
 //!    **Return value**
 //!
-//!    dot product: real(x)*real(y)+imag(x)*imag(y). For real numbers it is just the product.
+//!    dot product. For values it is just the product.
```
Comment (jfalcou) [inline include/eve/module/core/regular/dot.hpp:57]: I thought we decided that dot should go ? Let's discuss it for a later PR

File: include/eve/module/core/regular/frexp.hpp [hunk 6]
```diff
@@ -40,27 +40,36 @@ struct frexp_t : elementwise_callable<frexp_t, Options, pedantic_option, raw_opt
 //!   @code
 //!   namespace eve
 //!   {
-//!      template< eve::floating_value T >  eve::zipped<T,T> frexp(T x) noexcept;           //1
-//!      template< eve::floating_value T >  eve::zipped<T,T> frexp[pedantic](T x) noexcept; //2
+//!      // Regular overload
+//!      constexpr auto frexp(floating_value auto x)                          noexcept; // 1
+//!
+//!      // Lanes masking
+//!      constexpr auto frexp[conditional_expr auto c](floating_value auto x) noexcept; // 2
```
Comment (jfalcou) [inline include/eve/module/core/regular/frexp.hpp:47]: zipped<T,T> was correct

File: include/eve/module/core/regular/zip.hpp [hunk 7]
```diff
@@ -60,36 +60,25 @@ namespace eve
...
-  //!
-  //!     template<product_type Target, scalar_value... Ts>
-  //!     auto operator()(as<Target> t, Ts... parts) const noexcept; //2
-  //!
-  //!     template<simd_value... Ts>
-  //!     auto operator()(Ts... parts) const noexcept;               //3
-  //!
-  //!     template<product_type Target, simd_value... Ts>
-  //!     auto operator()(as<Target> t, Ts... parts) const noexcept; //4
+  //!      constexpr auto zip(scalar_value auto... parts)               noexcept; //1
```
Comment (jfalcou) [inline include/eve/module/core/regular/zip.hpp:63]: Target still need to be templated

File: include/eve/module/core/regular/sort.hpp [hunk 8]
```diff
@@ -68,25 +68,22 @@ namespace eve
 //!   @code
 //!   namespace eve
 //!   {
-//!     template <simd_value T, typename Less>
-//!     T sort(T x, Less less);               // (1)
-//!
-//!     template <simd_value T>
-//!     T sort(T x);                          // (2)
+//!      // Regular overloads
+//!      constexpr auto sort(value auto  x);                    noexcept; // 1
```
Comment (jfalcou) [inline include/eve/module/core/regular/sort.hpp:72]: It returns T, stop overusing auto

File: include/eve/module/core/regular/safe.hpp [hunk 9]
```diff
@@ -11,13 +11,15 @@
 
 namespace eve
 {
+// TODO DOC
+// is this decorator still in use ? Perhaps must be in kyosu
```
Comment (jfalcou) [inline include/eve/module/core/regular/safe.hpp:15]: it is used by load and store etc

File: include/eve/module/core/regular/none.hpp [hunk 10]
```diff
@@ -29,38 +29,30 @@ namespace eve
...
-//!      template< eve::value T >
-//!      bool none(T x) noexcept;
+//!      // Regular overloads
+//!      constexpr auto none(logical_value auto x)                                      noexcept; // 1
+//!      constexpr auto none(top_bits auto t)                                           noexcept; // 1
 //!
-//!      template< eve::top_bits M > //1
-//!      bool none(M m) noexcept;    //2
+//!      // Lanes masking
+//!      constexpr auto none[conditional_expr auto c](/* any of the above overloads */) noexcept; // 2
```
Comment (jfalcou) [inline include/eve/module/core/regular/none.hpp:37]: It returns bool

File: include/eve/module/core/regular/minimum.hpp [hunk 11]
```diff
@@ -29,31 +29,28 @@ namespace eve
...
-//!     * `x` :  [argument](@ref eve::value).
+//!     * `x`: [argument](@ref eve::value).
+//!     * `c`: [Conditional expression](@ref conditional_expr) masking the operation.
+//!     * `m`: [Logical value](@ref logical) masking the operation.
 //!
-//!    **Return value**
+//!   **Return value**
 //!
-//!    The scalar minimal value.
+//!     1. The scalar minimal value.
```
Comment (jfalcou) [inline include/eve/module/core/regular/minimum.hpp:49]: The minimal value of all lanes

File: include/eve/module/core/regular/maximum.hpp [hunk 12]
```diff
@@ -29,31 +29,28 @@ namespace eve
...
-//!     * `x` :  [argument](@ref eve::value).
+//!     * `x`: [argument](@ref eve::value).
+//!     * `c`: [Conditional expression](@ref conditional_expr) masking the operation.
+//!     * `m`: [Logical value](@ref logical) masking the operation.
 //!
-//!    **Return value**
+//!   **Return value**
 //!
-//!    The scalar maximal value.
+//!     1. The scalar maximal value.
```
Comment (jfalcou) [inline include/eve/module/core/regular/maximum.hpp:49]: the maximal value of all lanes

File: include/eve/module/core/regular/logical_not.hpp [hunk 13]
```diff
@@ -46,7 +46,7 @@ namespace eve
 //!   @code
 //!   namespace eve
 //!   {
-//!      eve::as_logical_t<T> logical_or(auto value x) noexcept;
+//!      constexpr auto logical_not(auto logical_value x) noexcept;
```
Comment (jfalcou) [inline include/eve/module/core/regular/logical_not.hpp:49]: STOP.PUTTING.AUTO.EVERYWHERE

**Review by jfalcou — CHANGES_REQUESTED** [review]
WTH with auto ???

File: include/eve/module/core/regular/bit_select.hpp (hunk 3 above)
Comment (jtlap) [reply include/eve/module/core/regular/bit_select.hpp:46]: done

File: include/eve/module/core/regular/safe.hpp (hunk 9 above)
Comment (jtlap) [reply include/eve/module/core/regular/safe.hpp:15]: ok line 15 suppressed

File: include/eve/module/core/regular/minimum.hpp (hunk 11 above)
Comment (jtlap) [reply include/eve/module/core/regular/minimum.hpp:49]: done

File: include/eve/module/core/regular/maximum.hpp (hunk 12 above)
Comment (jtlap) [reply include/eve/module/core/regular/maximum.hpp:49]: done

**Review by jfalcou — APPROVED** [review]

**Pull request merged**